hackathon_site/event/serializers.py

```diff
@@ -32,6 +32,36 @@ class Meta:
             "e_signature",
             "team",
         )
+        read_only_fields = ("id", "team", "acknowledge_rules", "e_signature")
+
+    def update(self, instance: Profile, validated_data):
+        if not validated_data:
```

Comment:
I had to raise a custom error because modifying read only fields/having an empty request doesn't automatically throw an error

Reply:
That's fine actually, the empty request part is valid since this is a patch request (meaning partial update). You can remove this error.

I didn't realize that submitting read only fields didn't throw an error though 🤔. Maybe it's easier to have a quick test case that just asserts what the read only fields (`Serializer.read_only_fields`) are on the serializer directly, so that nobody accidentally changes them.

Reply:
what do you mean by on the serializer directly?

Reply:
I mean add something in the test case like:

```
def test_readonly_serializer_fields(self):
    self.assertEqual(CurrentTeamOrderListView.read_only_fields, ("id", "team", "acknowledge_rules", "e_signature"))
```

You can put that in the test class for the view.

We know that DRF will use the read_only_fields properly, we just want a test to make sure nobody accidentally changes it without considering the implications. 

You can go ahead and remove this `if not validate_data` check as well, since I guess it conforms to the semantics of a PATCH request as it was.